Tests/test_file_mpo.py

```diff
@@ -293,16 +293,18 @@ def test_save_all() -> None:
             assert_image_similar(im, im_reloaded, 30)
 
     im = Image.new("RGB", (1, 1))
-    im2 = Image.new("RGB", (1, 1), "#f00")
-    im_reloaded = roundtrip(im, save_all=True, append_images=[im2])
-
-    assert_image_equal(im, im_reloaded)
-    assert isinstance(im_reloaded, MpoImagePlugin.MpoImageFile)
-    assert im_reloaded.mpinfo is not None
-    assert im_reloaded.mpinfo[45056] == b"0100"
-
-    im_reloaded.seek(1)
-    assert_image_similar(im2, im_reloaded, 1)
+    for colors in (("#f00",), ("#f00", "#0f0")):
+        append_images = (Image.new("RGB", (1, 1), color) for color in colors)
```

Comment:
```suggestion
        append_images = [Image.new("RGB", (1, 1), color) for color in colors]
```
This generator is being consumed in the line below, so we have nothing to iterate over in the `for` loop -- hence no coverage. Let's use a list.

Reply:
Thanks for catching that!